fuel-tx/src/transaction/consensus_parameters/gas_price_metadata.rs

```diff
@@ -0,0 +1,42 @@
+use core::num::NonZeroU64;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub struct GasPriceMetadata {
+    pub new_exec_gas_price: u64,
+    pub min_exec_gas_price: u64,
+    pub exec_gas_price_change_percent: u16,
+    pub l2_block_fullness_threshold_percent: u8,
+    // TODO:We don't need this after we implement
+    // https://github.com/FuelLabs/fuel-core/issues/2481
+    pub gas_price_factor: NonZeroU64,
+    pub min_da_gas_price: u64,
+    pub max_da_gas_price: u64,
+    pub max_da_gas_price_change_percent: u16,
```

Comment:
We only need this field, all other fields shouldn't be part of the consensus

Reply:
i want to disagree with this. while keeping just this one field is nice to ensure compatibility with all other operators, we have to make sure that all parameters with which the block producer determines the gas price (p, d, gas price factor etc) is included in the consensus parameters for transparency sake

Reply:
We don't know what algorithm we want to use for the gas price on consensus level. Right now, only one entity - the block producer, decides how the gas price is calculated. On the consensus level, we just agree that gas price shouldn't increase more than X %.

It is done for flexibility and to avoid setting the algorithm in stone.

Reply:
if that is the case then perhaps we should version this field in the consensus parameters. what do you think?